src/main/osd/osd_elements.c

```diff
@@ -1017,6 +1017,42 @@ static void osdElementPower(osdElementParms_t *element)
     tfp_sprintf(element->buff, "%4dW", getAmperage() * getBatteryVoltage() / 10000);
 }
 
+static void osdElementRcChannels(osdElementParms_t *element)
+{
+    const uint8_t xpos = element->elemPosX;
+    const uint8_t ypos = element->elemPosY;
+
+    for (int i = 0; i < OSD_RCCHANNELS_COUNT; i++) {
+        if (osdConfig()->rcChannels[i] >= 0) {
+            int decimals = osdConfig()->rcChannelDecimals[i];
+            int data = constrain(rcData[osdConfig()->rcChannels[i]] + osdConfig()->rcChannelOffsets[i], 0, 9999);
+            char fmtbuf[6];
+            if (decimals == 0) {
+                tfp_sprintf(fmtbuf, "%4d", data); // Formatted right justified
+                displayWrite(element->osdDisplayPort, xpos, ypos + i, fmtbuf);
+            } else {
+                tfp_sprintf(fmtbuf, "%04d", data); // Formatted right justified
```

Comment:
We really need to look into switching to https://github.com/mpaland/printf/blob/master/printf.c and slay the float printing dragon for good.

Reply:
One problem with `%f` is that ARM EABI passes floating point value in ... as `double` and we don't want to pull in double libraries. It can be solved with auxiliary function, but it's a bit ugly.

Otherwise I have modified version in my (commercial) project, I can prepare PR (possibly with thread/isr safe `debug(fmt, ...)`)

Reply:
Yes please @ledvinap!